### ISSUE TITLE
Upload all changed files together every x minutes

### ISSUE THREAD

**Issue by aero4** [nextcloud/desktop#287]
### Feature request
Is it possible to collect all file changes and uploading changed files together every `x` minutes (time  based)?
The problem is that if I sync a programming-directory via the nextcloud client, it is uploaded everytime I compile or save something...
Further details to my issue see: https://help.nextcloud.com/t/is-there-a-possibility-to-add-a-delay-between-filechange-and-upload/31418

That feature would stress the cloud storage less and would save my internet volume...

**Comment by tomdereub** [comment]
Yes, I too would like the possibility to configure this, and it would be more like "sync every 60 minutes". I work sometimes on big files (100-500 Mo), and I don't want it to be synchronised every time I save the file. For the moment, the only solution to avoid that is to manually pause synchronisation.

**Comment by nicolnt** [comment]
Could this be on a per-file and a per-folder basis please?

It could still be relevant to let the rest sync normally when updated for safety reasons.
The current behavior (sync on file change) would stay unchanged, and the user would have to edit this for each file or directory he or she wants a bigger delay.
Thanks :)

**Comment by tomdereub** [comment]
Any news about this feature request ?
In my opinion, doing this on a per-folder basis would be quite complicated to integrate and to use.
But 2 things could be (I hope) quite simple to implement : 
- add a parameter in the parameter tab of nextcloud, with a delay between 2 sync : x minutes. By default 0, means realtime sync (actual behaviour).
- when all syncs are suspended, add an option in the right-clic menu : sync all once. I mean : make a full sync and then stay suspended.
It would be really helpful to lighten internet traffic and server load in big organizations, and to easily control traffic on limited connections.

**Comment by OlafBr** [comment]
I would also like to have this feature:

- my banking program backup its db files when closing it - this takes a few seconds. But the nextcloud client immediately starts to sync the files - and then the backup fails.
- or you get an email and drag-drop the attachment to a folder. Then I always rename the file to something like yyyy-mm-dd_Name.txt but sync already startet and the windows explorer does a refresh and the renaming fails.

This feature would be really very helpful...

**Comment by mgallien** [comment]
I am not convinced there is value in delaying synchronization for a sync client.
You might have power outages or whatever and you would loose data that is not synced yet
